internal/nix/profile.go

```diff
@@ -311,3 +255,35 @@ func ProfileRemove(profilePath, nixpkgsCommit, pkg string) error {
 func AllowUnfreeEnv() []string {
 	return append(os.Environ(), "NIXPKGS_ALLOW_UNFREE=1")
 }
+
+type manifest struct {
+	Elements []struct {
+		Priority int `json:"priority"`
+	} `json:"elements"`
+}
+
+func readManifest(profilePath string) (manifest, error) {
+	data, err := os.ReadFile(filepath.Join(profilePath, "manifest.json"))
+	if os.IsNotExist(err) {
+		return manifest{}, nil
+	} else if err != nil {
+		return manifest{}, err
+	}
+
+	var m manifest
+	return m, json.Unmarshal(data, &m)
```

Comment:
I'm glad flakes changed the manifest from `.nix` to `.json` so we can do this :)